resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/v2/DriverPodKubernetesCredentialsMounter.scala

```diff
@@ -0,0 +1,173 @@
+      sparkConfWithCredentialLocations.set(KUBERNETES_SUBMIT_OAUTH_TOKEN, "<present_but_redacted>")
+    }
+    sparkConfWithCredentialLocations
+  }
+
+  override def createCredentialsSecret(): Option[Secret] = {
+    val allSecretData = resolveSecretData(
+        maybeUserSpecifiedMountedClientKeyFile,
+        submitterLocalDriverPodKubernetesCredentials.clientKeyDataBase64,
+        DRIVER_CREDENTIALS_CLIENT_KEY_SECRET_NAME) ++
+      resolveSecretData(
+        maybeUserSpecifiedMountedClientCertFile,
+        submitterLocalDriverPodKubernetesCredentials.clientCertDataBase64,
+        DRIVER_CREDENTIALS_CLIENT_CERT_SECRET_NAME) ++
+      resolveSecretData(
+        maybeUserSpecifiedMountedCaCertFile,
+        submitterLocalDriverPodKubernetesCredentials.caCertDataBase64,
+        DRIVER_CREDENTIALS_CA_CERT_SECRET_NAME) ++
+      resolveSecretData(
+        maybeUserSpecifiedMountedOAuthTokenFile,
+        submitterLocalDriverPodKubernetesCredentials.oauthTokenBase64,
+        DRIVER_CREDENTIALS_OAUTH_TOKEN_SECRET_NAME)
+    if (allSecretData.isEmpty) {
+      None
+    } else {
+      Some(new SecretBuilder()
+        .withNewMetadata().withName(s"$kubernetesAppId-kubernetes-credentials").endMetadata()
+        .withData(allSecretData.asJava)
+        .build())
+    }
+  }
+
+  override def mountDriverKubernetesCredentials(
+      originalPodSpec: PodBuilder,
+      driverContainerName: String,
+      credentialsSecret: Option[Secret]): PodBuilder = {
+    credentialsSecret.map { secret =>
+      originalPodSpec.editSpec()
+        .addNewVolume()
+          .withName(DRIVER_CREDENTIALS_SECRET_VOLUME_NAME)
+          .withNewSecret().withSecretName(secret.getMetadata.getName).endSecret()
+          .endVolume()
+        .editMatchingContainer(new ContainerNameEqualityPredicate(driverContainerName))
+          .addNewVolumeMount()
+            .withName(DRIVER_CREDENTIALS_SECRET_VOLUME_NAME)
+            .withMountPath(DRIVER_CREDENTIALS_SECRETS_BASE_DIR)
+            .endVolumeMount()
+          .endContainer()
+        .endSpec()
+    }.getOrElse(originalPodSpec)
+  }
+
+  private def resolveSecretLocation(
+    mountedUserSpecified: Option[String],
+    valueMountedFromSubmitter: Option[String],
+    mountedCanonicalLocation: String): Option[String] = {
+    mountedUserSpecified.orElse(valueMountedFromSubmitter.map( _ => {
+      mountedCanonicalLocation
+    }))
+  }
+
+  private def resolveSecretData(
+      mountedUserSpecified: Option[String],
+      valueMountedFromSubmitter: Option[String],
+      secretName: String): Map[String, String] = {
+    mountedUserSpecified.map { _ => Map.empty[String, String]}
+        .getOrElse {
+      valueMountedFromSubmitter.map { valueBase64 =>
+        Map(secretName -> valueBase64)
+      }.getOrElse(Map.empty[String, String])
+    }
+  }
+
+  private implicit def augmentSparkConf(sparkConf: SparkConf): OptionSettableSparkConf = {
```

Comment:
The `private` modifier ensures that the implicit scope only applies to this class.